user_sync/rules.py

```diff
@@ -1092,9 +1092,6 @@ def write_stray_key_map(self):
                 if not secondary_count:
                     fieldnames.append('umapi')
                 secondary_count += 1
-        # None sorts before strings, so sorting the keys in the map
-        # puts the primary umapi first in the output, which is handy
-        for umapi_name in sorted(self.stray_key_map.keys()):
```

Comment:
@vossen-adobe don't we still need `umapi_name` here?  I don't think we want to remove the loop, just remove the call to `sorted()`

Reply:
I combined it with the loop above instead of iterating over the same list twice

Reply:
I don't see a commit for that change in this PR

Reply:
It's the same as removing 1095-1097 - once the sorted line is gone, 1097 is identical to the for loop above it, so I just removed the line and comments.  There's no other changes to commit

Reply:
```
        for umapi_name in self.stray_key_map:
            if umapi_name != PRIMARY_UMAPI_NAME and self.get_stray_keys(umapi_name):
                if not secondary_count:
                    fieldnames.append('umapi')
                secondary_count += 1
        **for umapi_name in sorted(self.stray_key_map.keys()): <- just dropped this line**
            for user_key in self.get_stray_keys(umapi_name):
                id_type, username, domain = self.parse_user_key(user_key)
                umapi = umapi_name if umapi_name else ""
                if secondary_count:
                    row_dict = {'type': id_type, 'username': username, 'domain': domain, 'umapi': umapi}
                else:
                    row_dict = {'type': id_type, 'username': username, 'domain': domain}
                rows.append(row_dict)
```

Reply:
Thanks, I see it now.  However we need that first loop to execute separately so we have a total `secondary_count` before we execute a the loop to write the CSV.  Otherwise we'll have undefined behavior because that variable governs which columns are included in the output.

Reply:
Yes - that makes sense. Let me fix that!